### ISSUE TITLE
copy/transfer SocketListener

### ISSUE THREAD

**Issue by bmeck** [jasnell/sockets-api#7]
this likely is niche and not needed but it seems like if SocketListener is able to be copied between threads that handling initial marshalling of the connection/data could be shared.

**Comment by jasnell** [comment]
Yeah, I take this approach with the QUIC implementation currently. Specifically, a QUIC `Endpoint` can be created in one thread and shared with another. All of the actual i/o for the `Endpoint` takes place in the original owning thread but both are able to use the `Endpoint` concurrently. It works really well and was pretty straightforward to implement. It'll be a bit different with TCP sockets, of course. The listener can receive the initial connection on one thread and transfer ownership of that to another. In Node.js, we currently don't have the ability to transfer `uv` handles to other event loop instances so we'd either need to implement that first or use the an alternative that is similar to what I've done in the QUIC impl.

**Comment by bmeck** [comment]
yup, i'm talking about the overall design allowing for it, I don't think anything currently prevents this from being a follow on

**Comment by jasnell** [comment]
To draw this out a bit more, in Node.js, the specific options are:

1. Make the uv handles transferable across threads. Destination thread takes over ownership.
2. Just hand off the underlying fd once it's opened, set up a new uv handle on the receiving thread.  Destination thread takes over ownership.
3. Sending thread retains ownership and performs the basic i/o, cooperates with the destination thread that is using the connection.
